.github/workflows/ci.yml

```diff
@@ -72,6 +72,16 @@ jobs:
     - uses: actions/checkout@v4
       with:
         persist-credentials: false
+    - uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+          $VCPKG_DEFAULT_BINARY_CACHE
+        key: ${{ runner.os }}-cargo-stable-${{ hashFiles('Cargo.lock') }}
```

Comment:
Sorry to only notice this now but you probably want a separate `restore-key` here to use the cache even if `Cargo.lock` changes otherwise every dependency bump will completely invalidate the cache.

Reply:
Personally I'm fine with invalidating the cache a little bit too often rather than too rarely. Also, Cargo.lock is ideally only touched very rarely. I.e. I don't want to set up dependabot or run `cargo update` too regularly. It's just for testing and development purposes.